CMakeLists.txt

```diff
@@ -91,6 +91,15 @@ if (FIBER_ENABLE_FFTADVMPI)
 endif()
 
 find_package(MPI REQUIRED)
+if (NOT MPI_FOUND OR (MPI_FOUND AND NOT MPI_ROOT))
+  message(WARNING "MPI package not properly found")
+  if (DEFINED ENV{MPI_ROOT})
```

Comment:
If we found `MPI_ROOT` then we don't need `ENV{MPI_ROOT}`. Why not make these cases separate? Test for `MPI_ROOT` and if found then use it. Then if necessary, try `ENV{MPI_ROOT}` and use that if available. And if all fails just quit with an error. What MPI has this problem? Should we submit an upstream issue?

Reply:
I do not know if the issue comes from MPI itself. It is just everytimes I try to link, it cannot set MPI correctly and I end up wth a compilation error where libmpi cannot be found.
As far as I remember, MPI_FOUND was false.

Reply:
My experience with the modules for find_package is that they are not very well documented. In case of MPI, I don't see in the documentation if find_package(MPI) is searching and setting MPI_ROOT. Looks like we check if the user set it in their environment. In that case I am wondering why do we need it - and do we use it here (have to look at the code later).

Reply:
I have really bad experience with find_package.
I just noticed that MPI_ROOT or MPI_DIR should be needed.
So, in the case where find_package fails, I propose to handle it kind of manually.

In the end, you are right wondering why we need it. I do not know.

Reply:
If `MPI_FOUND` is undefined then you don't have MPI. And obtaining `MPI_ROOT` location is an optional feature, not a requirement. You might be asking for too much if you need MPI and the location of its headers and libraries. Note that `libmpi` is not what MPI requires for the name of MPI implementation. And it's usually multiple libraries that you need with additional linker settings to wire in the right path. You should be using `mpicc` and `mpic++` wrappers.

Reply:
Ok, maybe I was not pedantic enough. The compilation reported:
```
[ 70%] Linking C executable test3D_CPU_C2C                                                                                                                                                                                                                                                
/usr/bin/ld: cannot find -lmpi                                                                                                                                                                                                                                                            
collect2: error: ld returned 1 exit status                                                                                                                                                                                                                                                
make[2]: *** [benchmarks/test3D_CPU_C2C] Error 1                                                                                                                                                                                                                                          
make[1]: *** [benchmarks/CMakeFiles/test3D_CPU_C2C.dir/all] Error 2                                                                                                                                                                                                                       
make: *** [all] Error 2
```
So, it may need more, but at least needs this.

Reply:
Normally find_package(MPI) sets the library path to where libmpi is located, it works in all systems I've tried. This error shows to me when not loading MPI. Sebastien, since the CMake of the library was working in your system when compiling with  FFTW and heFFTe, perhaps it has to do with the way your system integrates the compilation to support MPI with Fortran?
Also, a quick solution would be to compile in Summit and Spock, the find_package(MPI) does the right job there, and there is where we need to perform experiments.

Reply:
No Alan, it was not working either when I was working on FFTW and heFFTe.

I use cmake 3.22 as well.

Reply:
The current CMake does compile and works fine with FFTW. heFFTe, and other integrated backends on Saturn, otherwise the CI would be failing.

Reply:
The point here is not to say this does not work at all.
I am raising a case where it does not.
Now, if we do not want to address this, we can pause this PR or even remove it and I will patch the library when needed.